### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by alexandrlonski**
cinema_DB

I created a database, connected to it and added tables
Ticket: https://trello.com/c/Vzu3QDBW

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,7 @@
+PORT=7011
+DB_NAME=online_cinema
+DB_USER=postgres
+DB_PASSWORD=lonski1206
+DB_HOST=localhost
+DB_PORT=5432
+SECRET_KEY=sekret_key_321
```
Comment (yavorovsky) [inline .env:7]: .env file should be ignored, here is a secure information

File: controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+const jwt = require("jsonwebtoken");
+const { User, Basket } = require("../models/models");
+
+const generateJwt = (id, email, role) => {
+  return jwt.sign({ id, email, role }, process.env.SECRET_KEY, {
+    expiresIn: "24h",
+  });
+};
+
+class UserController {
```
Comment (yavorovsky) [inline controllers/userController.js:12]: Why do you need the class here if you use it once and have no usage "this" here ??

File: middleware/ErrorHandlingMiddleware.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const ApiError = require("../error/ApiError");
```
Comment (yavorovsky) [inline middleware/ErrorHandlingMiddleware.js:1]: How about adding "logger"? If you run the server constantly and something was broken a week ago, how would you detect it?

Comment (yavorovsky) [reply middleware/ErrorHandlingMiddleware.js:1]: Take a look at "morgan" and stuff like this

File: controllers/userController.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  async login(req, res, next) {
+    const { email, password } = req.body;
+    const user = await User.findOne({ where: { email } });
+    if (!user) {
+      return next(ApiError.internal("User is not found"));
+    }
+    let comparePassword = bcrypt.compareSync(password, user.password);
+    if (!comparePassword) {
+      return next(ApiError.internal("Wrong password"));
```
Comment (yavorovsky) [inline controllers/userController.js:37]: Maybe would rather have a file with constants for messages?

File: controllers/userController.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  async registration(req, res, next) {
+    const { email, password, role } = req.body;
+    if (!email || !password) {
+      return next(ApiError.badRequest("Incorrect email or password"));
+    }
+    const candidate = await User.findOne({ where: { email } });
+    if (candidate) {
+      return next(ApiError.badRequest("User with this email already exists"));
+    }
+    const hashPassword = await bcrypt.hash(password, 5);
```
Comment (yavorovsky) [inline controllers/userController.js:22]: hash info (5 in this case) and etc. should be hidden in .env file

File: error/ApiError.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+class ApiError extends Error {
+  constructor(status, message) {
+    super();
+    this.status = status;
+    this.message = message;
+  }
+
+  static badRequest(message) {
```
Comment (yavorovsky) [inline error/ApiError.js:8]: Nice approach 👍

File: error/ApiError.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    return new ApiError(404, message);
```
Comment (yavorovsky) [inline error/ApiError.js:9]: Take a look at https://www.npmjs.com/package/http-status-codes

File: index.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+app.use("/api", router);
+
+// Error Handling
+app.use(errorHandler);
+
+const start = async () => {
+  try {
+    await sequelize.authenticate(); // Connect with database
+    await sequelize.sync(); // Sync with database
+    app.listen(PORT, () => console.log(`Server started omn port ${PORT}`));
```
Comment (yavorovsky) [inline index.js:23]: on*

File: middleware/checkRoleMiddleware.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
+const jwt = require("jsonwebtoken");
+
```
Comment (yavorovsky) [inline middleware/checkRoleMiddleware.js:2]: why checkRoleMiddleware duplicates authMiddleware ?

**Review by yavorovsky — CHANGES_REQUESTED** [review]

File: index.js (hunk 8 above)
Comment (alexandrlonski) [reply index.js:23]: fixed

File: .env (hunk 1 above)
Comment (alexandrlonski) [reply .env:7]: fixed

File: controllers/userController.js (hunk 5 above)
Comment (alexandrlonski) [reply controllers/userController.js:22]: fixed

File: middleware/checkRoleMiddleware.js (hunk 9 above)
Comment (alexandrlonski) [reply middleware/checkRoleMiddleware.js:2]: fixed

File: controllers/userController.js (hunk 2 above)
Comment (alexandrlonski) [reply controllers/userController.js:12]: +

File: controllers/userController.js (hunk 4 above)
Comment (alexandrlonski) [reply controllers/userController.js:37]: +

File: error/ApiError.js (hunk 7 above)
Comment (alexandrlonski) [reply error/ApiError.js:9]: +

**Pull request merged**